### PR TITLE
JAMES-3440 Task to populate EmailQueryView

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by Arsnael** [comment]
Why is the first commit a fixup?

**Comment by chibenwa** [comment]
> Why is the first commit a fixup?

Mistake. It should be a standalone commit. I will fix on next rebase.

File: server/protocols/webadmin/webadmin-jmap/src/test/java/org/apache/james/webadmin/data/jmap/PopulateEmailQueryViewRequestToTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("taskId", is(taskId))
+            .body("type", is("PopulateEmailQueryViewTask"))
+            .body("additionalInformation.runningOptions.messagesPerSecond", is(20));
+    }
+
+    @Test
+    void recomputeUserShouldUpdateProjection() throws Exception {
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-jmap/src/test/java/org/apache/james/webadmin/data/jmap/PopulateEmailQueryViewRequestToTaskTest.java:258]: recomputeUser ?

File: server/protocols/webadmin/webadmin-jmap/src/test/java/org/apache/james/webadmin/data/jmap/PopulateEmailQueryViewRequestToTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        with()
+            .basePath(TasksRoutes.BASE)
+            .get(taskId + "/await");
+
+        assertThat(view.listMailboxContent(messageId.getMailboxId(), Limit.from(12)).collectList().block())
+            .containsOnly(messageId.getMessageId());
+    }
+
+    @Test
+    void recomputeUserShouldBeIdempotent() throws Exception {
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-jmap/src/test/java/org/apache/james/webadmin/data/jmap/PopulateEmailQueryViewRequestToTaskTest.java:278]: idem

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
```
[16ac3bbc810131f62745981742656f1ba40cd28c] [INFO] Running org.apache.james.mailrepository.cassandra.CassandraMailRepositoryWithFakeImplementationsTest
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0 s <<< FAILURE! - in org.apache.james.mailrepository.cassandra.CassandraMailRepositoryWithFakeImplementationsTest
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR] org.apache.james.mailrepository.cassandra.CassandraMailRepositoryWithFakeImplementationsTest  Time elapsed: 0 s  <<< ERROR!
[16ac3bbc810131f62745981742656f1ba40cd28c] java.lang.NoClassDefFoundError: Could not initialize class org.apache.james.backends.cassandra.DockerCassandraSingleton
[16ac3bbc810131f62745981742656f1ba40cd28c] 
[16ac3bbc810131f62745981742656f1ba40cd28c] [INFO] 
[16ac3bbc810131f62745981742656f1ba40cd28c] [INFO] Results:
[16ac3bbc810131f62745981742656f1ba40cd28c] [INFO] 
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR] Errors: 
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR]   CassandraMailRepositoryCountDAOTest » NoClassDefFound Could not initialize cla...
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR]   CassandraMailRepositoryKeysDAOTest » NoClassDefFound Could not initialize clas...
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR]   CassandraMailRepositoryMailDAOTest » NoClassDefFound Could not initialize clas...
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR]   CassandraMailRepositoryTest » ExceptionInInitializer
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR]   CassandraMailRepositoryUrlStoreTest » NoClassDefFound Could not initialize cla...
[16ac3bbc810131f62745981742656f1ba40cd28c] [ERROR]   CassandraMailRepositoryWithFakeImplementationsTest » NoClassDefFound Could not...
```
Little guice binding problem i believe :)

**Comment by chibenwa** [comment]
test this please

Cassandra container timeout.

The CI looks tired to me ;-)

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**